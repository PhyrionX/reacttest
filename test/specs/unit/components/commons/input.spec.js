import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import Input from '../../../../../src/components/commons/input';

describe('<Input />', () => {
  it('should mount empty component', () => {
    const { asFragment } = render(<Input />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount without label, with type, id and placeholder', () => {
    const { asFragment } = render(<Input type="text"
        id="testId"
        placeholder="testPlaceholder" />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount with label, type, id and placeholder', () => {
    const { asFragment } = render(<Input label="testLabel"
        type="text"
        id="testId"
        placeholder="testPlaceholder" />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount with label, type, id, placeholder and error', () => {
    const { asFragment } = render(<Input label="testLabel"
        type="text"
        id="testId"
        error
        placeholder="testPlaceholder" />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount with value and onChange', () => {
    const { asFragment } = render(<Input label="testLabel"
        type="text"
        id="testId"
        error
        value="testValue"
        placeholder="testPlaceholder"
        onChange={() => {}} />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should mount without onKeyPress and fire event', () => {
    const mockFunctionOnKeyPress = jest.fn();
    const mockFunctionOnChange = jest.fn();
    const { getByLabelText } = render(<Input label="testLabel"
        type="text"
        id="testId"
        error
        value="testValue"
        placeholder="testPlaceholder"
        onChange={ mockFunctionOnChange } />);
    
    const input = getByLabelText("input");

    fireEvent.keyPress(input, {key: 'Enter', keyCode: 13 });
    expect(mockFunctionOnKeyPress).toHaveBeenCalledTimes(0); 
  })


  it('should mount onKeyPress and fire event', () => {
    const mockFunctionOnKeyPress = jest.fn();
    const mockFunctionOnChange = jest.fn();
    const { getByLabelText } = render(<Input label="testLabel"
        type="text"
        id="testId"
        error
        value="testValue"
        placeholder="testPlaceholder"
        onKeyPress={ mockFunctionOnKeyPress }
        onChange={ mockFunctionOnChange } />);
    
    const input = getByLabelText("input");

    fireEvent.keyPress(input, {key: 'Enter', keyCode: 13 });
    expect(mockFunctionOnKeyPress).toHaveBeenCalled(); 
  })
})