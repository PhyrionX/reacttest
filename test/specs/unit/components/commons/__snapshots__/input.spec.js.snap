// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should mount empty component 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <input
      aria-label="input"
      class="form-control "
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<Input /> should mount with label, type, id and placeholder 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      for="testId"
    >
      testLabel
    </label>
    <input
      aria-label="input"
      class="form-control "
      id="testId"
      placeholder="testPlaceholder"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<Input /> should mount with label, type, id, placeholder and error 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      for="testId"
    >
      testLabel
    </label>
    <input
      aria-label="input"
      class="form-control is-invalid"
      id="testId"
      placeholder="testPlaceholder"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<Input /> should mount with value and onChange 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      for="testId"
    >
      testLabel
    </label>
    <input
      aria-label="input"
      class="form-control is-invalid"
      id="testId"
      placeholder="testPlaceholder"
      type="text"
      value="testValue"
    />
  </div>
</DocumentFragment>
`;

exports[`<Input /> should mount without label, with type, id and placeholder 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <input
      aria-label="input"
      class="form-control "
      id="testId"
      placeholder="testPlaceholder"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
